### PR TITLE
Issue/327

### PR DESCRIPTION
**Pull request by minborg**
This fixes numerous JSON bugs

### CODE REVIEW

**Comment by minborg** [comment]
@peter-k-lawrey could you please have a look at the build errors introduced in 8ec4e26f67d11cbd4ff6c4437c72b4f9bef13c70 ? 

https://teamcity.chronicle.software/buildConfiguration/OpenHFT_ChronicleWire_PullRequestsLinux/621895?

**Review by RobAustin — APPROVED** [review]
@minborg - the changes look ok to me, can you confirm new or existing tests cover these changes, as it's difficult to see the test coverage in the review tool.

**Comment by minborg** [comment]
We have added around 20 tests each running with types/no types

File: src/main/java/net/openhft/chronicle/wire/JSONWire.java [hunk 1]
```diff
@@ -482,13 +496,46 @@ void readTypeDefinition(StringBuilder sb) {
...
-/*        if (!type.getName().startsWith("java.lang.")) {
-            // exclude most classes as soon as possible
-            return false;
-        }*/
         return type == Integer.class || type == Long.class || type == Float.class ||
                 type == Double.class || type == Short.class || type == Character.class ||
                 type == Byte.class || type == Boolean.class || type == Void.class;
     }
 
+/*
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/wire/JSONWire.java:509]: commented-out code?

Comment (minborg) [reply src/main/java/net/openhft/chronicle/wire/JSONWire.java:509]: Yepp. Missed that... Thanks

**Pull request merged**